include/aligator/modelling/centroidal/wrench-cone.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "aligator/core/function-abstract.hpp"
+
+namespace aligator {
+
+/**
+ * @brief This residual implements the "ice cream" friction cone for a
+ * centroidal model with state \f$x = (c, h, L) \f$.
+ *
+ * @details Considering an unilateral contact k exerting 3D force u,
+ * the residual returns a two-dimension array with first component
+ * equal to \f$ \epsilon - u_z \f$ (strictly positive normal force
+ * condition) and second component equal to * \f$ u_{x,y}^2 -
+ * \mu^2 * u_{z}^2 \f$ (non-slippage condition) with \f$ \epsilon
+ * \f$ small threshold and \f$ \mu \f$ friction coefficient.
+ */
+
+template <typename Scalar> struct WrenchConeDataTpl;
+
+template <typename _Scalar>
+struct WrenchConeResidualTpl : StageFunctionTpl<_Scalar> {
+
+public:
+  using Scalar = _Scalar;
+  ALIGATOR_DYNAMIC_TYPEDEFS(Scalar);
+  using Base = StageFunctionTpl<Scalar>;
+  using BaseData = typename Base::Data;
+  using Data = WrenchConeDataTpl<Scalar>;
+
+  WrenchConeResidualTpl(const int ndx, const int nu, const int k,
+                        const double mu, const double L, const double W)
```

Comment:
:warning: `W` is a half-width and `L` is a half-length in this context. With these short names future travelers will assume they are the width and length, effectively quadrupling the contact area.

Suggestions:

- Document the function to describe each `@param` with a short sentence :pray:
- Use longer variable names like `half_length` and `half_width`?
- Keep the names but internally store halved values with proper names, e.g. `half_width_` or `hW_`?

Reply:
I applied your suggestions, thanks!